go-apps/meep-platform-ctrl/server/user_authentication.go

```diff
@@ -50,6 +237,35 @@ func uaLoginUser(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	// Start user session
+	sandboxName, err, errCode := startSession(username, w, r)
+	if err != nil {
+		log.Error(err.Error())
+		http.Error(w, err.Error(), errCode)
+		return
+	}
+
+	// Prepare response
+	var sandbox dataModel.Sandbox
+	sandbox.Name = sandboxName
+
+	// Format response
+	jsonResponse, err := json.Marshal(sandbox)
+	if err != nil {
+		log.Error(err.Error())
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	// Send response
+	w.Header().Set("Content-Type", "application/json; charset=UTF-8")
+	w.WriteHeader(http.StatusOK)
+	fmt.Fprint(w, string(jsonResponse))
+}
+
+// Retrieve existing user session or create a new one
+func startSession(username string, w http.ResponseWriter, r *http.Request) (sandboxName string, err error, code int) {
+
 	// Get existing session by user name, if any
 	sessionStore := pfmCtrl.sessionMgr.GetSessionStore()
 	session, err := sessionStore.GetByName(username)
```

Comment:
Username is possibly not unique now that several providers are supported

Reply:
This is fixed in an upcoming PR. Provider-specific sessions & user database entries will be used to prevent this issue.